hazelcast/src/main/java/com/hazelcast/map/impl/mapstore/writebehind/StoreWorker.java

```diff
@@ -83,11 +82,31 @@ public StoreWorker(MapStoreContext mapStoreContext, WriteBehindProcessor writeBe
         NodeEngine nodeEngine = mapServiceContext.getNodeEngine();
         InternalPartitionService partitionService = nodeEngine.getPartitionService();
         this.partitionCount = partitionService.getPartitionCount();
+        this.executionService = nodeEngine.getExecutionService();
     }
 
 
+    public synchronized void start() {
+        stopped = false;
+        run();
```

Comment:
shouldn't this just schedule an execution instead of running on the calling thread?